docs/migrate-from-V4-to-V5.md

```diff
@@ -0,0 +1,93 @@
+
+We suggest to you to clone the example projects and insert your code into the new projects.
+
+There are a number of changes that you will need to make to your code to move from V4 to V5.
+
+And we have included a list of the main changes below:
+
+## Moving from Create-React-App to Vite
+
+1. Migrate your .env file and rename all variables in your .env file from `REACT_APP_` to `VITE_` and remove the `REACT_APP_` prefix.
+2. Vite requires all files with JSX to be named with the `.jsx` extension. You will need to rename the files with JSX from `.js` files to `.jsx` files.
+3. In your JSX files, if you have imports with a .js extension, you will need to remove the extension.
+4. We have kept all the ports the same as in V4. This means that you will not need to change the ports.
+5. Any imports that used './React-Automation-Studio/...' will need to be changed to 'React-Automation-Studio/...'.
+## Updating the backend and frontend packages
+
+1. The pvServer uses EPICS 7.0.7 and Python 3.12.1, if you have any customizations to the pvServer, you will need to update the pvServer to use the latest version of EPICS and Python.
+2. The frontend uses the latest version of React and React-Router-Dom.
+    We were stuck on React-Router-Dom 5.3.3 and we have now moved to React-Router-Dom 6.21.3. You will need to update your code to use the latest versions of these packages.
+    
+    We followed the migration guide at https://reactrouter.com/docs/en/v6/getting-started/migrating.
+    This impacted us in the following ways. There was a clash in ReactApp/src/Routes.js name with the new React-Router-Dom 6.21.3. We had to rename the file to ReactApp/src/AppRoutes.jsx.
+    We needed to update the code in the file to use the new version of React-Router-Dom. This also impacted the ProtectedRoute component:
+
+    Previously in the Routes.js we had:
+
+    ```jsx
+    <BrowserRouter>
+       <Switch>
+          <ProtectedRoute
+             path="/ArchiverDataViewerDemo"
+             component={ArchiverDataViewerDemo}
+          />
+          ... more routes
+       </Switch>
+    </BrowserRouter>
+    ```
+
+    Now in the AppRoutes.jsx we have:
+
+    ```jsx
+    <BrowserRouter>
+       <Routes>
+          <Route
+             path="/AlarmHandlerDemo"
+             element={
+                <ProtectedRoute>
+                   <AlarmHandlerDemo />
+                </ProtectedRoute>
+             }
+          />
+          ... more routes
+       </Routes>
+    </BrowserRouter>
+    ```
+    We also had to update the code in the ReactApp/src/components/SystemComponents/Login.Login.jsx to use the new version of React-Router-Dom. If you used a custom version then please update it to use the new version of React-Router-Dom.
+
+## Multistage builds
+    We have introduced multistage builds for the docker compose files. 
+    This means that you will need to update your docker-compose files to use the new multistage builds.
```

Comment:
I know this is a copy from other RAS project but it might be helpful if we provide overview of images/docker files and intended use. Maybe, this should be included into  separated document and reference it from here so one does not need to read through every bit to get a picture of the latest structure of the solution.

Reply:
ok  i'll make a note  an lets update the docs in V5.0.1